code/game/objects/objs.dm

```diff
@@ -48,6 +48,11 @@ var/global/list/reagents_to_log = list(FUEL, PLASMA, PACID, SACID, AMUTATIONTOXI
 	if(src in processing_objects)
 		processing_objects -= src
 
+	if(integratedpai)
+		integratedpai.forceMove(get_turf(src))
```

Comment:
There are multiple ways to get `qdel()` called on something.
Which ones should/shouldn't result in the destruction of an integrated pAI?